Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -1426,7 +1469,89 @@ -(void) textViewDidProgrammaticallyUpdate:(UITextView *)textView {
     [self resetAuxiliaryState];
 }
 
+- (void)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text isInsertion:(BOOL)isInsertion {
+    BOOL isCreatingOrStartingMention = self.state == HKWMentionsStartDetectionStateCreatingMention || [self isStringControlCharacter:text];
+    if (!isCreatingOrStartingMention) {
+        // Only implement Korean mentions fix when creating or starting mentions
+        // We will revisit a longer term alignment of using the fix in non-mentions cases, after we ramp the mentions fix
+        return;
+    }
+    self.suppressSelectionChangeNotifications = YES;
+    __strong __auto_type parentTextView = self.parentTextView;
+    unichar precedingChar = [parentTextView characterPrecedingLocation:(NSInteger)range.location];
+
+    if (self.nextInsertionShouldBeIgnored) {
+        self.nextInsertionShouldBeIgnored = NO;
+        self.previousSelectionRange = textView.selectedRange;
+        self.previousTextLength = [textView.text length];
+        self.suppressSelectionChangeNotifications = NO;
+        return;
+    }
+
+    if (isInsertion && [text length] == 1) {
+        // Inserting a character
+        [self advanceStateForCharacterInsertion:[text characterAtIndex:0]
+                             precedingCharacter:precedingChar
+                                       location:range.location];
+    }
+    else if (!isInsertion) {
+        // Deleting text
+        NSString *toDeleteString = text;
+        if (range.length == 0) {
+            // At the beginning, and delete was tapped
+            self.suppressSelectionChangeNotifications = NO;
+            return;
+        }
+        else if (range.length == 1) {
+            // Deleting a single character
+            [self advanceStateForCharacterDeletion:precedingChar
+                                  deletedCharacter:[toDeleteString characterAtIndex:0]
+                                          location:range.location];
+        }
+        else {
+            // Deleting multiple characters
+            [self korean_mentions_fix_advanceStateForStringDeletionAtRange:range
+                                                             deletedString:toDeleteString
+                                                        precedingCharacter:precedingChar];
+            // Reset the selection range
+            textView.selectedRange = range;
+        }
+    }
+    else if (isInsertion && [text length] > 1) {
```

Comment:
nit, would be nice to group this up with the other isInsertion case

Reply:
added to ticket for clean ups of this kind of code copied over from the non-lix implementation